src/Symfony/Component/Cache/Adapter/DoctrineDbalAdapter.php

```diff
@@ -375,8 +375,7 @@ private function getServerVersion(): string
             return $this->serverVersion;
         }
 
-        // The condition should be removed once support for DBAL <3.3 is dropped
-        $conn = method_exists($this->conn, 'getNativeConnection') ? $this->conn->getNativeConnection() : $this->conn->getWrappedConnection();
+        $conn = $this->conn->getNativeConnection();
         if ($conn instanceof ServerInfoAwareConnection) {
```

Comment:
Hmm, this does not really make sense. If we call `getNativeConnection()`, we get the native handle of the PHP extension used, e.g. a `mysqli` instance or an `sqlsrv` resource. Those objects/resources won't ever implement `ServerInfoAwareConnection`. Something's off here.

Reply:
I guess you are right. Looks like we need to fix it ina way similar to #52715.